### PR TITLE
Fix lint and add lint + prettier to CI 

### PR DESCRIPTION
**Pull request by ZhangYiJiang**
I've noticed while there is a lint and prettier command configured, they are not actually enforced by CI, so some of the code has drifted. Most can be automatically or manually fixed but a few places where the linter has picked up genuine bugs has been left due to lack of context needed to fix them.  

This PR tries to bring the code up to lint as much as possible conservatively and add them to CI so they don't drift in the future. 

The one big bug that it found is an import cycle between RunGraphNode and the globalRivetNodeRegistry global. There's no easy way to reconcile this other than making the global registry lazy by putting it behind a getter, but that's more code alteration than seems good for a lint fix PR so I'll delegate that to a future PR to fix.

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -4,7 +4,7 @@ module.exports = {
     es2021: true,
   },
   root: true,
-  extends: ['standard-with-typescript', 'plugin:react/recommended'],
```
Comment (ZhangYiJiang) [inline .eslintrc.cjs:7]: This plugin only appears to be used in app, so configuring it here seems to be a mistake

File: packages/core/src/model/nodes/HttpCallNode.ts [hunk 2]
```diff
@@ -211,7 +211,9 @@ export class HttpCallNodeImpl extends NodeImpl<HttpCallNode> {
     const method = getInputOrData(this.data, inputs, 'method', 'string');
     const url = getInputOrData(this.data, inputs, 'url', 'string');
 
+    // TODO: Use URL.canParse when we drop support for Node 18
```
Comment (ZhangYiJiang) [inline packages/core/src/model/nodes/HttpCallNode.ts:214]: Not sure if we have an official Node support policy

File: packages/core/src/plugins/google/google.ts [hunk 3]
```diff
@@ -74,6 +74,7 @@ export type ChatCompletionChunk = {
   model: string;
 };
 
+/* eslint-disable @typescript-eslint/naming-convention */
```
Comment (ZhangYiJiang) [inline packages/core/src/plugins/google/google.ts:77]: Not sure if it is better to leave this to match API naming convention or rename to match Rivet/JS naming convention

**Comment by ZhangYiJiang** [comment]
The new checks thingy is pretty nice although it does mean the existing warnings will appear on every PR until they are resolved (but it also means new warnings will show up more prominently which is nice)

**Comment by ZhangYiJiang** [comment]
I'll merge this for now so we can also run lint on the new PRs coming in, the warnings should be fixed later

File: .eslintrc.cjs (hunk 1 above)
Comment (abrenneke) [reply .eslintrc.cjs:7]: It was intentional I think, because it didn't harm anything, but can't remember much beyond that

Comment (ZhangYiJiang) [reply .eslintrc.cjs:7]: The package.json entries for these were also missing (it was causing errors when running the linter)

**Comment by abrenneke** [comment]
Thanks for doing this!

**Pull request merged**